ext/js/language/ar/language-arabic.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {basicTextPreprocessorOptions} from '../text-preprocessors.js';
+
+/** @type {import('language/language-arabic').ArabicLanguageDescriptor} */
+export const descriptor = {
+    name: 'Arabic',
+    iso: 'ar',
+    exampleText: 'قَرَأَ',
+    textPreprocessors: {
+        removeDiacritics: {
```

Comment:
This can likely also be used for Farsi so maybe it should be moved to ../text-preprocessors.js?